680 design and implement a new landing page UI

igawaclawska opened the pull request:
# Changes
## New Hero Layout:
Redesigned the structure to include a dynamic grid for language selection on the hero section

| smartphone | tablet |
|----------|----------|
| <kbd>![mobile-hero](https://github.com/user-attachments/assets/73159470-8d1e-4389-ac30-2bf63eee83f5)</kbd> | <kbd>![tablet-hero](https://github.com/user-attachments/assets/687f4082-1a46-4f23-bac2-b5a2c0061cce)</kbd> |

| desktop (not full-screen) | desktop |
|----------|----------|
| <kbd>![desktop-not-full-screen](https://github.com/user-attachments/assets/1f7f212d-18fc-45c5-939e-f8e3eb49d077)</kbd> | <kbd>![desktop](https://github.com/user-attachments/assets/e7f453ce-35d2-4768-9d32-6df90aa11f53)</kbd> |


## Dynamic Flag Integration:
Added flag icons for each language on the `LandingPage` and `LanguageModal` using the `<DynamicFlagImage/>` component.

| Landing Page | Language Modal |
|----------|----------|
| <kbd>![desktop](https://github.com/user-attachments/assets/e7f453ce-35d2-4768-9d32-6df90aa11f53)</kbd> | <kbd><img width="1200" alt="language-modal" src="https://github.com/user-attachments/assets/0fea8b88-77b6-4ac3-b30e-3a89db796cde" /></kbd>|

## Routing Speed improvement
Enhanced routing speed among the Landing Page, Login, Registration, and other auth and onboarding pages by using routing methods provided by `react-router-dom` for navigation **within** React.

## Next
Further improvements and work on the "How does it work?" section

mircealungu commented:
Looks amazing @igawaclawska !

mircealungu commented:
And I love the Language modal even more :)

igawaclawska commented:
Hi @mircealungu, I'm happy to hear you like it! 
Should I continue the implementation (the "How does it work?" section) from this branch or branch out and wait for the merge? :)

mircealungu commented:
maybe one more thing: i'm having a bit of a hard time focusing on the message on the left hand side of the page, given that there's so much "color" on the right hand side. 

<img width="1344" alt="image" src="https://github.com/user-attachments/assets/e89ea78e-29fc-4938-b545-4ad61ad68198" />

does this make sense? 
does anybody else have this feeling? (e.g., @tfnribeiro) 

is there something that we can do about it?

tfnribeiro commented:
I think I agree with @mircealungu - but it's a bit hard for me to imagine how it's like for a new user, for the most part I feel like the fact that the buttons are interactable are also taking a bit away from the focus of the text on the left.

So, what I am thinking is that the Text shares with the Get started and the languages could instead be presented below, similar to what we do already if the page is thinner:

![{7BC82903-BA32-4D99-A2AD-477C63C6AF8C}](https://github.com/user-attachments/assets/e8ba7af9-a3d5-484c-81f1-259723fd836a)

Another thing to consider is that instead of showing all languages, we show the top 5 and the user can then scroll to see more. This would also solve the "problem" that some languages we list there aren't "active", so we could have the situation where someone signs up and they see a language that we aren't actively crawling.

mircealungu commented:
That’s a very good point Tiago about us not even crawling all the languages.
Should we update available_languages to only return languages that we are crawling?

On 3 Apr 2025 at 10.19 +0200, Tiago Ribeiro ***@***.***>, wrote:
>
> I think I agree with @mircealungu - but it's a bit hard for me to imagine how it's like for a new user, for the most part I feel like the fact that the buttons are interactable are also taking a bit away from the focus of the text on the left.
>
>
> So, what I am thinking is that the Text shares with the Get started and the languages could instead be presented below, similar to what we do already if the page is thinner:
>
>
> 7BC82903-BA32-4D99-A2AD-477C63C6AF8C.png (view on web)
>
>
> Another thing to consider is that instead of showing all languages, we show the top 5 and the user can then scroll to see more. This would also solve the "problem" that some languages we list there aren't "active", so we could have the situation where someone signs up and they see a language that we aren't actively crawling.
>
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.Message ID: ***@***.***>
>
> tfnribeiro left a comment (zeeguu/web#747)
> I think I agree with @mircealungu - but it's a bit hard for me to imagine how it's like for a new user, for the most part I feel like the fact that the buttons are interactable are also taking a bit away from the focus of the text on the left.
> So, what I am thinking is that the Text shares with the Get started and the languages could instead be presented below, similar to what we do already if the page is thinner:
> 7BC82903-BA32-4D99-A2AD-477C63C6AF8C.png (view on web)
> Another thing to consider is that instead of showing all languages, we show the top 5 and the user can then scroll to see more. This would also solve the "problem" that some languages we list there aren't "active", so we could have the situation where someone signs up and they see a language that we aren't actively crawling.
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.Message ID: ***@***.***>

tfnribeiro commented:
That's a good idea. We can do that, though there is no programmatic way of doing it... I guess for now we could just have a manual string?

mircealungu commented:
I think it’s to just hardcode the languages that we support for now.
On 3 Apr 2025 at 10.30 +0200, Tiago Ribeiro ***@***.***>, wrote:
>
> That's a good idea. We can do that, though there is no programmatic way of doing it... I guess for now we could just have a manual string?
>
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.Message ID: ***@***.***>
>
> tfnribeiro left a comment (zeeguu/web#747)
> That's a good idea. We can do that, though there is no programmatic way of doing it... I guess for now we could just have a manual string?
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.Message ID: ***@***.***>

mircealungu commented:
@igawaclawska - should we consider also separating the little description below the big text on their own lines to ease reading?

<img width="895" alt="image" src="https://github.com/user-attachments/assets/85124716-728a-43f4-a2f4-ad60203a69a7" />

<img width="1336" alt="image" src="https://github.com/user-attachments/assets/a64f0286-0f05-4f45-854e-0b6c6b36c263" />

mircealungu commented:
or maybe to attract more attention to the LHS, use a heart emoji? :) 

<img width="1317" alt="image" src="https://github.com/user-attachments/assets/e4155933-2cf4-4b66-803f-040c242c8f59" />

mircealungu commented:
also, "target language" vs "learning language'? target sounds a bit technical?

igawaclawska commented on src/components/TopNavLink.sc.js:
```diff
@@ -0,0 +1,65 @@
+import { Link } from "react-router-dom/cjs/react-router-dom";
+import { orange600, orange700, orange800 } from "./colors";
+
+const ListItem = styled.li`
```
Info is now added to both the TopNav and TopNavOption components.
It's a most recommended structure for the navigation as it gives good compatibility with assistive technologies (allows screen readers to announce how many elements are in the navigation)  🙂

<nav>
      <ul>
           <li><Link/></li>
           <li><Link/></li>
           <li><Link/></li>
      <ul>
<nav>

// https://www.w3.org/WAI/tutorials/menus/structure/#identify-menus
// https://www.w3.org/WAI/tutorials/menus/structure/#menu-representation

igawaclawska commented on src/components/TopNavLink.js:
```diff
@@ -0,0 +1,24 @@
+import * as s from "./TopNavLink.sc";
+
+export default function TopNavLink({
```
Renamed to `TopNavOption`

igawaclawska commented on src/components/TopNav.sc.js:
```diff
@@ -0,0 +1,39 @@
+import styled, { css } from "styled-components";
+
+const _navbarShared = css`
```
Renamed to `_topNavShared`

igawaclawska commented:
@mircealungu , @tfnribeiro 

## Regarding the updates to the hero design:

Our goal is to drive conversions, so it’s important that all elements of the hero, both visuals and text, direct the user’s attention towards the call-to-action button. I agree that we currently have multiple competing buttons. 

**Here are my thoughts on the proposed solutions:**

1. **Column layout**:  My favorite, and I am currently implementing this solution. I like the idea of a column layout with the language options placed below for all screens. This design naturally directs user attention from the text to the call-to-action button, and still allows users to convert through the language buttons if they choose not to click the main one.

2. **Decrease font size and weight of language buttons**: I suspect that the colorful flags are drawing attention, so decreasing the font size and weight may not be helpful enough.

3. **Leaving only flags**: Not everyone remembers flags, which might make it challenging for users to associate them with specific languages (additionally, the reason above).

4. **Separating the description**: This change alters the shape of the text block and visual flow. It could shift user focus toward the large text instead of the button (at least it happens to me). 

5. **Adding a heart icon**: The heart looks playful, but unfortunately, it shifts attention away from the call-to-action button.

igawaclawska commented:
@mircealungu , @tfnribeiro 

> That's a good idea. We can do that, though there is no programmatic way of doing it... I guess for now we could just have a manual string?

> I think it’s to just hardcode the languages that we support for now.

The Hero is now vertically aligned. Given that I am to experiment with a language carousel in further iterations and keep grid in this one, which languages should I hardcode to be displayed on the page?

mircealungu commented:
> @mircealungu , @tfnribeiro
> 
> ## Regarding the updates to the hero design:
> Our goal is to drive conversions, so it’s important that all elements of the hero, both visuals and text, direct the user’s attention towards the call-to-action button. I agree that we currently have multiple competing buttons.
> 
> **Here are my thoughts on the proposed solutions:**
> 
> 1. **Column layout**:  My favorite, and I am currently implementing this solution. I like the idea of a column layout with the language options placed below for all screens. This design naturally directs user attention from the text to the call-to-action button, and still allows users to convert through the language buttons if they choose not to click the main one.
> 2. **Decrease font size and weight of language buttons**: I suspect that the colorful flags are drawing attention, so decreasing the font size and weight may not be helpful enough.
> 3. **Leaving only flags**: Not everyone remembers flags, which might make it challenging for users to associate them with specific languages (additionally, the reason above).
> 4. **Separating the description**: This change alters the shape of the text block and visual flow. It could shift user focus toward the large text instead of the button (at least it happens to me).
> 5. **Adding a heart icon**: The heart looks playful, but unfortunately, it shifts attention away from the call-to-action button.

I agree with your analysis @igawaclawska :)

igawaclawska commented on src/landingPage/LandingPage.js:
```diff
@@ -1,69 +1,88 @@
   }
 
+  function handleLanguageSelect(language) {
+    LocalStorage.setLearnedLanguage(language);
```
Now the selected language is no longer passed to the next page (`LanguagePreferences.js`) via Local Storage. It uses search params instead. 

I have decided to keep the Local Storage integration between the `LanguagePreferences.js` page and the next page, `CreateAccount.js`, to preserve the data when the user proceeds to `CreateAccount.js` and decides to go back to change something. If I had used params in this case, the selection would have been lost.

igawaclawska commented on src/landingPage/LandingPageTopNav.js:
```diff
@@ -0,0 +1,35 @@
+      <TopNavLink to="/log_in">{strings.login}</TopNavLink>
+      <TopNavLink
+        callToAction
+        onClick={clearLearnedLanguage}
```
Exactly!

igawaclawska commented on src/pages/_pages_shared/Logo.js:
```diff
@@ -1,5 +1,19 @@
+    <svg
+      width={size ? size : "2.25rem"}
+      height={size ? size : "2.25rem"}
+      viewBox="0 0 71 71"
```
**A little bit about the component itself:** 
This component contains the code I extracted from the SVG elephant logo by opening it in a text editor and pasting the code into a component (I only turned width, height, and fill into props, and added `aria-hidden` as it's a decorative image). I believe this can make the logo more versatile for scaling/resizing/recoloring, as we don't need to keep various versions, just changing props is enough.

**About the viewBox:**
To my understanding, this represents a viewport for the image with coordinates. 
Would you prefer if I added a link about this in the code? 
https://www.digitalocean.com/community/tutorials/svg-svg-viewbox

igawaclawska commented:
@mircealungu , @tfnribeiro 
I have addressed all the questions and implemented the changes. 

Additionally, I have modified how we fetch the list of available languages (`systemLanguages`). Previously, we would fetch them from api separately on the `LandingPage.js`, `LanguagePreferences.js`, and `LanguageSettings.js` pages, but since it's shared data, I have implemented it as a context - this way it's only fetched once when the app loads and becomes available for all the pages above.

mircealungu commented on src/pages/onboarding/CreateAccount.js:
```diff
@@ -161,7 +161,7 @@ export default function CreateAccount({ handleSuccessfulLogIn }) {
           setUserDetails(userInfo);
           saveUserInfoIntoCookies(userInfo);
-          redirect("/select_interests");
+          history.push("/select_interests");
```
Thanks Iga! This is such a good find! Can I write a LinkedIn recomendation for you? :)
